### PR TITLE
lirc: use it only for user-configured remotes

### PR DESCRIPTION
**Pull request by HiassofT**
Now that all platforms have been migrated to kernel / rc-core for IR remote handling we can finally remove all bits that were needed to support IR remotes out-of-the-box via usersapce lircd and use lircd only when users want to support odd remotes via a custom lircd.conf file.

- lirc startup is now only controlled by the presence of a lircd.conf file in /storage/.config. Separate lirc enable/disable in LE settings can be dropped (I'll PR that separately)
- no lircd.conf files will be preinstalled. The currently provided conf files are either superseded by ir-keytable configs (eg MCE remotes) or obsolete (eg xbox-dvd which worked only with the removed lirc_xbox driver).

So basically this means LE will ship with an empty lirc config and lirc won't be enabled unless a user manually installs a conf file. This also removes the dreaded "double clicks", as lirc can't be enabled accidentally any longer.

@Kwiboo @codesnake could you check this is fine on Wetek Hub and Odroid C2?

### CODE REVIEW

**Comment by codesnake** [comment]
I've checked and WeTek Hub IR remote works OOB with this PR.

**Comment by chewitt** [comment]
@HiassofT if this is complete please remove WIP and we'll merge it in, thanks.

**Comment by HiassofT** [comment]
@Raybuntu reported that meson-ir (WetekHub and Odroid C2) isn't quite in shape yet, eg MCE remotes don't work properly

Problem is that kernel protocol decoders need a trailing space / timeout to decode signals and meson-ir driver doesn't generate that.

We'll have a look if there's an easy fix for that. If not my plan B is to keep the old MCE and xbox lircd.conf files as templates in the image so people on WH/C2 can easily copy them over to /storage/.config and stick with lirc until meson-ir is finally fixed.

**Comment by HiassofT** [comment]
We managed to get amlogic devices / meson-ir into shape, last bit of the puzzle was changing the ir-keytable default config to provide a good out-of-the-box experience on WH/C2 - see PR #2424. When that PR is merged this one is good to go as well.
